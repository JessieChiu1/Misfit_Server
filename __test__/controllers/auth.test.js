const app = require("../../src/app")
const { MongoMemoryServer } = require("mongodb-memory-server");
const supertest = require("supertest");
const mongoose = require("mongoose")
const dotenv = require("dotenv/config");

const userPayload = {
    username: "jessie",
    password: "password",
}

const invalidPayload = {
    username: "wrongUsername",
    password: "wrongPassword",
}

describe("auth.js", () => {

    beforeAll(async () => {
        mongoServer = await MongoMemoryServer.create()
        await mongoose.connect(mongoServer.getUri())
    });

    afterAll(async () => {
        await mongoose.disconnect()
        await mongoose.connection.close()
    });

    describe("signup", () => {
        describe("given userPayload", () => {
            it("should return 200 and jwt", async () => {
                const response = await supertest(app)
                    .post("/api/v1/auth/signup")
                    .send(userPayload)
                // expectation
                expect(response.status).toBe(200)
                expect(response.body).toHaveProperty("token");
                expect(typeof response.body.token).toBe("string")
                // snapshot
                expect(response.body).toMatchSnapshot({
                    "token": expect.any(String)
                })
            })
        })
    })

    describe("login", () => {
        describe("given an invalid userPayload", () => {
            it("should return 400", async () => {
                // act
                const response = await supertest(app)
                    .post("/api/v1/auth/login")
                    .send(invalidPayload)
                // expectation
                expect(response.status).toBe(400)
                // snapshot
                expect(response.body).toMatchSnapshot()
            })
        })
        describe("given a valid userPayload", () => {
            it("should return 400", async () => {
                // act
                const response = await supertest(app)
                    .post("/api/v1/auth/login")
                    .send(userPayload)
                // expectation
                expect(response.status).toBe(200)
                expect(response.body).toHaveProperty("token");
                expect(typeof response.body.token).toBe("string")
                // snapshot
                expect(response.body).toMatchSnapshot({
                    "token": expect.any(String)
                })
            })
        })
    })
})
