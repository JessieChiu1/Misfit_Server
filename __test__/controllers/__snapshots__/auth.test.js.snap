// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth.js login given a valid userPayload should return 400 1`] = `
{
  "token": Any<String>,
}
`;

exports[`auth.js login given an invalid userPayload should return 400 1`] = `
{
  "message": "Incorrect Username or Password",
}
`;

exports[`auth.js signup given userPayload should return 200 and jwt 1`] = `
{
  "token": Any<String>,
}
`;
